### ISSUE TITLE
How to create linkML from OWL?

### ISSUE THREAD

**Issue by peiyaoli** [monarch-initiative/ontogpt#237]
Sorry, I am not familiar with knowledge graph things. I want to know can I create LinkML yaml from one OWL file? Like https://www.ebi.ac.uk/ols4/ontologies/bao BAO ontology from EBI?

**Comment by serenalotreck** [comment]
I'd like to second this! I want to use OntoGPT with the [Planteome ontologies](https://planteome.org/node/1)

**Comment by caufieldjh** [comment]
Hi @peiyaoli and @serenalotreck - it's certainly a good question and unfortunately made more complicated by the fact that an OWL ontology isn't quite doing the same thing as a LinkML schema.

One option, depending on your use case, is to use a relatively simple schema (like this one: https://github.com/monarch-initiative/ontogpt/blob/main/src/ontogpt/templates/human_phenotype.yaml) but use the OWL file as one of the annotators. It could be specified like this in the schema if you were trying to extract names of bioassays, for example:
```
  Bioassay:
    is_a: NamedEntity
    id_prefixes:
      - BAO
    annotations:
      annotators: bao.owl
      prompt: >- 
        the name of a bioassay or bioassay component, for example, bioavailability assay.
```

I believe you could even specify the annotator as `sqlite:bao.owl` to convert it to a more performant sqlite version.

The Plant Ontology is already in OBO Foundry (http://obofoundry.org/ontology/po.html) so the annotator in this case could be `sqlite:obo:po`. Same goes for components like the Plant Stress Ontology.

So that assumes that the terms in the ontology are what you're trying to extract and/or ground to. If you'd rather use the OWL more like a schema to develop a corresponding LinkML schema, you may be able to do so with the schema-automator tool (https://github.com/linkml/schema-automator) and something like this:

```
$ schemauto import-owl bao.owl
``` 

Though I'll note that when I try this with `bao_complete.owl` then I get an error:

```
Exception: Empty ontology in bao_complete.owl (note: ontologies must be in functional syntax)
```

If I do something like this with [robot](http://robot.obolibrary.org/) then I get a bit closer:
```
$ robot convert -i bao_complete.owl -o bao.ofn
$ schemauto import-owl bao.ofn
```

But BAO relies on a whole bunch of imports, so this process also doesn't go as planned.

I tried this with PSO too...and it _almost_ appears to work, until raising a mysterious TypeError. 

It _does_ work for PO itself.

Note that the output LinkML YAML will still identify classes by their IDs, like so:
```yaml
  PO_0025358:
    comments:
    - Found in some palms and other species growing in flooded habitats. Pneumatorhiza
      appear as small outgrowths on the surface of a prop root. Similar, but distinct
      from, pneumatophores.
    is_a: PO_0020121
    class_uri: obo:PO_0025358
```

and to make this into a useful OntoGPT schema, classes will require slots like `description` and `annotations`.

So I would suggest modifying an existing schema and using the OWL as an annotator rather than converting directly from OWL to LinkML. But it can be done!

**Comment by TribeDH** [comment]
Hi everyone, thanks for all the effort you're putting in this project! I'm working on the extraction of events and arguments in a different domain, the legal and juridical one. In this domain resources are really scarse, for example don't exist any predefined event schema. I only found a few ontologies which could be useful, like this one: https://github.com/PLN-FaMAF/legal-ontology-population or this one: https://github.com/essepuntato/allot
Do you think it's also possible to automatic convert this kind of ontologies to LinkML Schemas? It could be really nice to widen the range of possibilities that this project can handle.
Thank you very much,
Mathias.

**Comment by caufieldjh** [comment]
Thanks @TribeDH - I'm going to split your question into its own issue, though I see there are OWL representations of the two ontologies you reference so it's related.

**Comment by nlharris** [comment]
Can this be closed?

**Comment by caufieldjh** [comment]
Yes, addressed by #257
